### PR TITLE
add as_slice

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/redismodule.rs [hunk 1]
```diff
@@ -145,6 +145,13 @@ impl RedisString {
         Self::from_ptr(self.inner).map_err(|_| RedisError::Str("Couldn't parse as UTF-8 string"))
     }
 
+    pub fn as_slice<'a>(&self) -> &'a [u8] {
```
Comment (gavrie) [inline src/redismodule.rs:148]: No need for the explicit lifetime annotation, the compiler infers it just fine.

File: src/redismodule.rs [hunk 2]
```diff
@@ -145,6 +145,13 @@ impl RedisString {
...
+        let mut len: libc::size_t = 0;
+        let bytes = unsafe { raw::RedisModule_StringPtrLen.unwrap()(self.inner, &mut len) };
+
+        unsafe { slice::from_raw_parts(bytes.cast::<u8>(), len) }
```
Comment (gavrie) [inline src/redismodule.rs:152]: This duplicates the code in `to_string_lossy`; I suggest making `to_string_lossy` call this function instead.

**Review by gavrie — CHANGES_REQUESTED** [review]

File: src/redismodule.rs [hunk 3]
```diff
@@ -153,10 +160,7 @@ impl RedisString {
     ///
     /// Will panic if `RedisModule_StringPtrLen` is missing in redismodule.h
     pub fn to_string_lossy(&self) -> String {
-        let mut len: libc::size_t = 0;
-        let bytes = unsafe { raw::RedisModule_StringPtrLen.unwrap()(self.inner, &mut len) };
-        let bytes = unsafe { slice::from_raw_parts(bytes.cast::<u8>(), len) };
-        String::from_utf8_lossy(bytes).into_owned()
+        String::from_utf8_lossy(self.as_slice()).into_owned()
```
Comment (oshadmi) [inline src/redismodule.rs:163]: Can do the same also in `from_ptr`

**Review by oshadmi — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**